frontends/web/src/components/confirm/Confirm.tsx

```diff
@@ -15,24 +15,34 @@
  */
 
 import { Component, h } from 'preact';
+import Dialog from '../dialog/dialog';
 import { Button } from '../forms';
-import * as style from './Confirm.css';
 
-export default class Confirm extends Component {
-    state = {
-        message: '',
-        active: false,
-    }
+var confirmation: (message: string, callback: () => void) => void;
+
+interface State {
+    message: string;
+    active: boolean;
+}
+
+interface Confirm {
+    callback: (input: boolean) => void;
+}
 
-    componentDidMount() {
-        // @ts-ignore (override changes the method signature)
-        window.confirm = this.confirmOverride;
+class Confirm extends Component<{}, State> {
+    constructor(props) {
+        super(props);
+        confirmation = this.confirmation;
+        this.state = {
+            active: false,
+            message: '',
+        };
     }
 
-    confirmOverride = (message, callback) => {
+    confirmation = (message, callback) => {
```

Comment:
Types of `message` and `callback` will be implicitly `any` (which is not ideal, see above).